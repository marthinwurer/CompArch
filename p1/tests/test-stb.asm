; test the stb instruction
	.entry	start
;
	.org	0x100
one:	.word	0x00001
five:	.word	0x00005
neg1:	.word	0xfffff
neg5:	.word	0xffffb
;
d1:	.word	0
d2:	.word	0
;
	.org	0x300
start:	ldb	one
	stb	d1
	ldb	d2
	ldb	d1
;
	ldb	five
	stb	d1
	ldb	d2
	ldb	d1
;
	ldb	neg1
	stb	d1
	ldb	d2
	ldb	d1
;
	ldb	neg5
	stb	d1
	ldb	d2
	ldb	d1
;
	halt
