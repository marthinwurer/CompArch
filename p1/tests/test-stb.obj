100 1 00001
101 1 00005
102 1 fffff
103 1 ffffb
104 1 00000
105 1 00000
300 1 11100
301 1 13104
302 1 11105
303 1 11104
304 1 11101
305 1 13104
306 1 11105
307 1 11104
308 1 11102
309 1 13104
30a 1 11105
30b 1 11104
30c 1 11103
30d 1 13104
30e 1 11105
30f 1 11104
310 1 00000
300
