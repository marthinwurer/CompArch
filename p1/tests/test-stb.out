CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 300
300:  11100  LDB   100  A[00000]  B[00000]  XR[000]
301:  13104  STB   104  A[00000]  B[00001]  XR[000]
302:  11105  LDB   105  A[00000]  B[00001]  XR[000]
303:  11104  LDB   104  A[00000]  B[00000]  XR[000]
304:  11101  LDB   101  A[00000]  B[00001]  XR[000]
305:  13104  STB   104  A[00000]  B[00005]  XR[000]
306:  11105  LDB   105  A[00000]  B[00005]  XR[000]
307:  11104  LDB   104  A[00000]  B[00000]  XR[000]
308:  11102  LDB   102  A[00000]  B[00005]  XR[000]
309:  13104  STB   104  A[00000]  B[fffff]  XR[000]
30a:  11105  LDB   105  A[00000]  B[fffff]  XR[000]
30b:  11104  LDB   104  A[00000]  B[00000]  XR[000]
30c:  11103  LDB   103  A[00000]  B[fffff]  XR[000]
30d:  13104  STB   104  A[00000]  B[ffffb]  XR[000]
30e:  11105  LDB   105  A[00000]  B[ffffb]  XR[000]
30f:  11104  LDB   104  A[00000]  B[00000]  XR[000]
310:  00000  HALT       A[00000]  B[ffffb]  XR[000]

Machine Halted - HALT instruction executed

Simulated time 100 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
